cns/ipampool/v2/adapter.go

```diff
@@ -0,0 +1,36 @@
+package v2
+
+import (
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/crd/nodenetworkconfig/api/v1alpha"
+	v1 "k8s.io/api/core/v1"
+)
+
+var _ cns.IPAMPoolMonitor = (*adapter)(nil)
+
+type adapter struct {
+	nncSink chan<- v1alpha.NodeNetworkConfig
+	*Monitor
+}
+
+func (m *Monitor) AsV1(nncSink chan<- v1alpha.NodeNetworkConfig) cns.IPAMPoolMonitor {
```

Comment:
> Can it just implement the monitor interface

I specifically didn't want to do that. That architecture was created backwards and that interface is too broad.
 
In considering the long-term design, there is no reason for an interface to exist at all. The Monitor is a self-contained component that nothing else needs to directly communicate with. Thus the architecture of the v2 Monitor is totally decoupled from the rest of CNS - Pod events are pushed to an event stream (a channel), and the Monitor processes them asynchronously. No interface needs to exist, because nobody should call the Monitor synchronously - in other words data flow is unidirectional from "the thing that knows the scaling demand" (Pod Watcher) in to "the thing scales the IP pool" (Monitor), and that communication is the use case for a channel, not an interface.